src/ostorlab/serve_app/types.py

```diff
@@ -463,3 +465,119 @@ def resolve_agents(
 class AgentGroupsType(graphene.ObjectType):
     agent_groups = graphene.List(AgentGroupType, required=True)
     page_info = graphene.Field(common.PageInfo, required=False)
+
+
+class AssetScansMixin:
+    scans = graphene.List(
+        lambda: OxoScanType, last_only=graphene.Boolean(required=False)
+    )
+
+    def resolve_scans(self, info):
+        with models.Database() as session:
+            scans = session.query(models.Scan).filter(
+                models.Vulnerability.asset_id == self.id
+            )
+
+        return scans
+
+
+class AndroidStoreAssetType(graphene_sqlalchemy.SQLAlchemyObjectType, AssetScansMixin):
+    class Meta:
+        model = models.AndroidStore
+        fields = ("id", "package_name", "application_name")
+
+
+class AndroidStoreAssetInputType(graphene.InputObjectType):
+    package_name = graphene.String()
+    application_name = graphene.String()
+
+
+class IOSStoreAssetType(graphene_sqlalchemy.SQLAlchemyObjectType, AssetScansMixin):
+    class Meta:
+        model = models.IosStore
+        fields = ("id", "bundle_id", "application_name")
+
+
+class IOSStoreAssetInputType(graphene.InputObjectType):
+    bundle_id = graphene.String()
+    application_name = graphene.String()
+
+
+class AndroidFileAssetType(graphene_sqlalchemy.SQLAlchemyObjectType, AssetScansMixin):
+    class Meta:
+        model = models.AndroidFile
+        fields = ("id", "package_name", "path")
+
+
+class AndroidFileAssetInputType(graphene.InputObjectType):
+    file = scalars.Upload()
+    package_name = graphene.String()
+
+
+class IOSFileAssetType(graphene_sqlalchemy.SQLAlchemyObjectType, AssetScansMixin):
+    class Meta:
+        model = models.IosFile
+        fields = ("id", "bundle_id", "path")
+
+
+class IOSFileAssetInputType(graphene.InputObjectType):
+    file = scalars.Upload()
+    bundle_id = graphene.String()
+
+
+class UrlAssetType(graphene_sqlalchemy.SQLAlchemyObjectType, AssetScansMixin):
+    links = graphene.List(graphene.String, required=False)
+
+    class Meta:
+        model = models.Url
+        fields = ("id",)
```

Comment:
we dont need the actual url?

Reply:
The urls are resolved 2 lines after this code.